xcb.c

```diff
@@ -233,8 +235,12 @@ void grab_pointer_and_keyboard(xcb_connection_t *conn, xcb_screen_t *screen, xcb
         }
     }
 
-    if (tries <= 0)
+    if (tries <= 0) {
+        pam_state = STATE_PAM_FAIL;
+        redraw_screen();
+        usleep(2000000);
```

Comment:
Given that it's not obvious I'd prefer a comment here explaining why there's a sleep before exiting.